### ISSUE TITLE
Create customer account only with "-" sign

### ISSUE THREAD

**Issue by mnick73** [PrestaShop/PrestaShop#23446]
#### Describe the bug

In version 1.7.7.2 you can create a customer account using only the "-" sign for the first and last name. 

#### Expected behavior

I think a minimum of 3 characters is required.

#### Steps to Reproduce

1. **BO**>**SELL**>**Customers**>**Customers**
2. Click on **Add new customer** button
3. Set **-** as **First name** and as **Last name**
4. Save
5. See error => the customer is well saved

**Screenshots**

![Capture](https://user-images.githubusercontent.com/79770168/109401804-7f510b00-7959-11eb-8daf-9685210179cc.PNG)

#### Additional information

* PrestaShop version: 1.7.7.2
* PHP version: 7.2

**Comment by matks** [comment]
Hi @mnick73 there are many ways to create a customer in PrestaShop, can you describe which way you followed? In the `Steps to Reproduce` part

It would be best to simply complete the template 😉 so we have all the needed information

**Comment by mnick73** [comment]
hello,

I used the simple way on prestashop live demo site. See attached file,
[steps.zip](https://github.com/PrestaShop/PrestaShop/files/6056291/steps.zip)
 record with steps recorder.

**Comment by hibatallahAouadni** [comment]
Hello @mnick73 

I just edit your issue description, could you please confirm the steps to reproduce are right.
Please check and feedback.

Thanks!

**Comment by mnick73** [comment]
hello,

it is correct description.

thank you

**Comment by Hlavtox** [comment]
Be careful!! :D I personally know several asian people with names/surnames just 2 characters long, and apparently even 1 letter names exist. 👍
https://www.quora.com/Are-there-any-last-names-family-names-that-are-only-one-letter

**Comment by matks** [comment]
> Be careful!! :D I personally know several asian people with names/surnames just 2 characters long, and apparently even 1 letter names exist. 👍
> https://www.quora.com/Are-there-any-last-names-family-names-that-are-only-one-letter

Very nice point 😄

**Comment by mnick73** [comment]
a dynamic condition can be the solution. if one of the name fields contains one letter, the other must contain at least three.

**Comment by Hlavtox** [comment]
@mnick73 So what if my name is **Li E**?

**Comment by mnick73** [comment]
@Hlavtox it is not ok to be able to register someone by name  --

**Comment by Hlavtox** [comment]
@mnick73 That's true, but we must keep **-** enabled, because some names contain it.

**Comment by matks** [comment]
> a dynamic condition can be the solution. if one of the name fields contains one letter, the other must contain at least three.

I think I can find someone named Xu Li 😅

**Comment by mnick73** [comment]
@Hlavtox  agree 
@matks  agree

**Comment by hibatallahAouadni** [comment]
Hello everyone,

Thanks guys for the backup @matks @Hlavtox :rocket: 
Well @mnick73 I think you're convinced now :sweat_smile: 
So, I close the issue :wink: 

Thanks!

**Comment by mnick73** [comment]
I'm not convinced. 
I'm agree with

-keep - sign 
-accept first name or  last name with single letter

but 
it is not ok to be able to register someone by name -- or AA  and so on.

thanks anyway

**Comment by hibatallahAouadni** [comment]
Hello @mnick73 

> it is not ok to be able to register someone by name -- or AA and so on.

I agree with you :wink: 
So, what you suggest is to improve the validation, is that so?
If so, I will reopen the issue as an improvement :wink: 
Please feedback.

Thanks!

**Comment by mnick73** [comment]
Yes.

Thanks!

**Comment by hibatallahAouadni** [comment]
Hello @mnick73 

Ok, done :wink: 
Thanks for the suggestion, the Developer Team will take a look.
Please stay tuned !

Ping @PrestaShop/prestashop-core-developers do you think it's a doable to add some extra validation on these fields **First name** and **Last name** when creating a customer in BO?

**Comment by matks** [comment]
Yes, you'll find some rules already implement in https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/ConstraintValidator/CustomerNameValidator.php

But I dont know if it's really interesting to _block_ people who want to register as `-` . I mean... if they really want to input something with no sense, it's their problem 😅  ?

And it's an `improvement` indeed, not a bug. It would be a bug if we wanted to block such inputs.

**Comment by matks** [comment]
Hi @hibatallahAouadni from a developer POV there is no issue to block people who use `-` as firstname.

However, I dont know if it's a good investment.

Every time we add a new feature in the Core, whether it's something complex (like the Fuzzy Search) or something very simple like this (block people using `-`  as firstname) we add some code into PrestaShop. Over years, the quantity of code grows tremendously.

It's not only the quantity of code that grows: every behavior we add to the software, every capability we give to the software, whether it's something complex like the Fuzzy Search or something very simple like this issue, enters the list of "all the things the software does and we must maintain it". By "maintain it", I mainly mean "make sure it continues to work in all future versions, even though we modify the code around us daily".

PrestaShop is already big 😅  so I think we should start questioning whether "one more feature" is now worth the extra code, behavior and maintainance it will cost us. And as I said before, here, I'm really wondering if it's something worth it.

I personally think that, when you balance out the 2 options
- add some more code, which means 1 more behavior, which means more maintainance (devs that must make sure it's not broken in future versions and QA analysts that must verify it still works in future versions) to block customers that signup with "-" sign
- do nothing

I find the 1st option quite bad when said like this 😅 

I guess @PrestaShop/product-team should give his opinion too

**Comment by ldeprestashop** [comment]
You can limit name to 3 char min 

Exemple : Cambodge, "Y" is a first name.  

The only rule (for length) is 1 char min

**Comment by ldeprestashop** [comment]
> Yes, you'll find some rules already implement in https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/ConstraintValidator/CustomerNameValidator.php
> 
> But I dont know if it's really interesting to _block_ people who want to register as `-` . I mean... if they really want to input something with no sense, it's their problem 😅 ?
> 
> And it's an `improvement` indeed, not a bug. It would be a bug if we wanted to block such inputs.

Add new rules for block "-" don't add any value.

**Comment by ldeprestashop** [comment]
> Hi @hibatallahAouadni from a developer POV there is no issue to block people who use `-` as firstname.
> 
> However, I dont know if it's a good investment.
> 
> Every time we add a new feature in the Core, whether it's something complex (like the Fuzzy Search) or something very simple like this (block people using `-` as firstname) we add some code into PrestaShop. Over years, the quantity of code grows tremendously.
> 
> It's not only the quantity of code that grows: every behavior we add to the software, every capability we give to the software, whether it's something complex like the Fuzzy Search or something very simple like this issue, enters the list of "all the things the software does and we must maintain it". By "maintain it", I mainly mean "make sure it continues to work in all future versions, even though we modify the code around us daily".
> 
> PrestaShop is already big 😅 so I think we should start questioning whether "one more feature" is now worth the extra code, behavior and maintainance it will cost us. And as I said before, here, I'm really wondering if it's something worth it.
> 
> I personally think that, when you balance out the 2 options
> 
> * add some more code, which means 1 more behavior, which means more maintainance (devs that must make sure it's not broken in future versions and QA analysts that must verify it still works in future versions) to block customers that signup with "-" sign
> * do nothing
> 
> I find the 1st option quite bad when said like this 😅
> 
> I guess @PrestaShop/product-team should give his opinion too

We will think about ratio code complexity vs value added 

for this feature I think there is no value but as @matks  indicates we add complexity (for an almost null value gain)

**Comment by matks** [comment]
Just found https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/ about this topic

**Comment by florine2623** [comment]
Hello,

After evaluation, this issue will not be fixed by PrestaShop SA at this time. Please feel free to fix this issue and create a PR if you want to see it fixed.

Thank you

**Comment by boo-code** [comment]
Still reproducible on 9.2, and the thread already contains the reason it was never fixed: an agreed
problem with no agreed rule.

`CustomerNameValidator::PATTERN_NAME` is a blacklist. It excludes digits and a set of punctuation and
refuses a value that is only whitespace, but it never requires anything to be present, so a name built
only from the punctuation it does allow passes: `-`, `--`, `'`, `  -  `.

Registered through the front office form on a 9.2 shop, with the row read back from the database rather
than the page believed:

```
firstname=- lastname=-    redirected to /    ps_customer row created
```

The length rule proposed here was rejected correctly - one and two letter names exist, and the hyphen
has to stay allowed. But a length rule was never the only option. Requiring **one letter, in any
script** separates the two cases with no minimum at all:

```
-      --     - -     '      ''      -'-     "  -  "        rejected
E      Li     Xu Li   O'Brien   Jean-Luc   Ann-Marie        accepted
李     Þór    van der Berg    D. Smith                       accepted
```

One lookahead at the front of the existing pattern, `(?=\P{L}*\p{L})`, does it, and it subsumes the
`(?!\s*$)` guard it replaces since a string of spaces carries no letter either. Measured after the
change, same form, same method:

```
firstname=- lastname=-     stayed on /registration, no row created
firstname=Li lastname=E    redirected to /, row created
```

The second line is the case raised against a length rule on this thread, so the counter-example is
covered rather than argued about.

Worth noting for whoever picks this up: `Validate::isName()` is a separate and looser blacklist used for
other entities, and widening the same rule to all of them is a different change with a much larger blast
radius.
